### PR TITLE
Refactor MainActivity code and speaker activity view transition 

### PR DESCRIPTION
**Pull request by pranaypatel512**
fix: #1349

### CODE REVIEW

**Comment by harshitagupta30** [comment]
Hey @pranaypatel512, It's a good practice to first report an issue then create a PR for the same! Also in case of some enhancement you need to wait for the maintainer's approval, :+1:

**Comment by pranaypatel512** [comment]
@harshitagupta30 : Thanks for your quick replay. Yes, I forgot to create issue for this PR. Can I create issue for this and edit description by mentioning issue no?

**Comment by harshitagupta30** [comment]
@pranaypatel512 Yes, you can create issue for this and be descriptive about why you're refactoring the code. Is this PR contains two issues? Kindly create a PR for only one issue not more than that.

File: android/app/src/main/java/org/fossasia/openevent/adapters/SpeakersListAdapter.java [hunk 1]
```diff
@@ -124,10 +127,26 @@ public void onBindViewHolder(RecyclerViewHolder holder, final int position) {
...
+                    holder.speakerImage.setTransitionName(activity.getString(R.string.activity_speaker_image_trans));
+
+                    Pair<View, String> pair1 = Pair.create((View) holder.speakerImage, holder.speakerImage.getTransitionName());
+                    Pair<View, String> pair2 = Pair.create((View) holder.speakerName, holder.speakerName.getTransitionName());
+                    ActivityOptionsCompat options = ActivityOptionsCompat.makeSceneTransitionAnimation(activity, pair1, pair2);
+                    String speakerName = current.getName();
+                    Intent intent = new Intent(activity, SpeakerDetailsActivity.class);
+                    intent.putExtra(Speaker.SPEAKER, speakerName);
+                    activity.startActivity(intent,options.toBundle());
+
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/adapters/SpeakersListAdapter.java]: Please make sure that in SpeakerDetailActivity, the transition is postponed until the data is loaded.

See [this](https://www.androiddesignpatterns.com/2015/03/activity-postponed-shared-element-transitions-part3b.html)

This ensures that the UI will not look janky while transition happens

File: android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java [hunk 2]
```diff
@@ -486,27 +487,30 @@ public void run() {
...
-                appBarLayout.setExpanded(true, true);
+                appBarLayout.setExpanded(true, true);*/
+                replaceFragment(new TracksFragment(),getString(R.string.menu_tracks));
                 break;
             case R.id.nav_schedule:
                 atHome = false;
-                fragmentManager.beginTransaction()
+                replaceFragment(new ScheduleFragment(),getString(R.string.menu_schedule));
+
+                /*fragmentManager.beginTransaction()
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java]: You'd probably like to remove a code block, not just comment it out. You can always revert a commit but committing old redundant code blocks pollutes the code and needs someone else's attention to remove them later

**Comment by iamareebjamal** [comment]
Also, you should only make one change per pull request so that if they can be merged atomically :smile:

**Comment by arp95** [comment]
@pranaypatel512
please follow the commit guidelines for making a pr. and also as others have said prefer making one pr for one issue.

**Comment by pranaypatel512** [comment]
@harshitagupta30 @arp95 @iamareebjamal 
I am just closing this PR and will send single PR per issue. Thanks you to all for comments.

**Pull request closed without merging**